src/auth/auth.service.ts

```diff
@@ -0,0 +1,93 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import * as bcrypt from 'bcrypt';
+import { UserService } from '../user/user.service';
+import { RegisterDto } from '../user/dto/registerDto';
+import { UserDetails } from '../user/user.datails.interface';
+import { LoginDto } from '../user/dto/loginDto';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private userService: UserService,
+    private jwtService: JwtService,
+  ) {}
+
+  // just a basic has+salt the password method.
+  async hashPassword(password: string): Promise<string> {
```

Comment:
I think you should make all methods that are used only inside this class PRIVATE (like hashPassword, doesPasswordMatch etc) :)